src/query.rs

```diff
@@ -800,6 +799,7 @@ mod tests {
                 "abc": (inherit: "abc"),
                 "string": "literal_string",
                 "int": -30,
+                "list": [-30, -2, "abc"],
```

Comment:
Heterogeneous lists like this aren't something Trustfall supports. This test case might make future refactoring harder if e.g. the `TransparentValue` type changes in a way that disallows that at the type level — the person doing it will have to also fix this test case too. It's usually not a good idea to exceed the bounds of the spec, because if the implementation of the spec starts enforcing the bounds more strongly then breakage ensues.

I'd recommend removing the string here, and making a separate list entry with a string or two to test strings inside the list.